controllers/user.js

```diff
@@ -56,10 +76,39 @@ const makeAvatar = async (userId, filePath) => {
   return newFilename;
 };
 
+const emailVerification = async (verificationToken) => {
+  const user = await User.findOne({ verificationToken });
+  if (!user) {
+    throw new HttpError("User not found", 404);
+  }
+  if (user.verify) {
+    throw new HttpError("Verification has already been passed", 400);
+  }
+  user.verify = true;
+  // user.verificationToken = null; // usuwanie tokena po weryfikacji
```

Comment:
Tak, bo jest ustawione na required. Zobacz ten post https://stackoverflow.com/questions/51814875/set-required-mongoose-schema-type-to-null

generalnie trzeba sprawdzić czy verify jest true i jeśli tak to zmienic dynamicznie na nie required.